### PR TITLE
feat: Added basic Attribute support + documentation to Blend

### PR DESCRIPTION
**Pull request by nightcycle**

### CODE REVIEW

File: src/blend/src/Shared/Blend/Blend.lua [hunk 1]
```diff
@@ -548,6 +575,48 @@ function Blend.Children(parent, value)
...
+]=]
+function Blend.Attributes(inst, value)
+	assert(typeof(inst) == "Instance", "Bad instance")
+	return Blend.ComputedPairs(value, function(attributeName, attributeValue, innerMaid)
+		local function getValue(v)
+			if type(v) == "table" then
+				if ValueObject.isValueObject(v) then
+					return getValue(v.Value)
+				elseif Brio.isBrio(v) then
+					return getValue(v:GetValue())
```
Comment (Quenty) [inline src/blend/src/Shared/Blend/Blend.lua:607]: nit: Should probably keep lifetime of attribute application to the brio lifetime. I think you can do 

```lua
innerMaid[attributeName] = v:GetDiedSignal():Connect(function()
    inst:SetAttribute(attributeName, attributeValue)
end)
```

You'll also need to set
innerMaid[attributeName] = nil

Before you do each item.


This should only occur if we emit a brio with another brio in it, so we could also remove the `if Brio.isBrio` condition entirely. Up to you.

File: src/blend/src/Shared/Blend/Blend.lua [hunk 2]
```diff
@@ -548,6 +575,48 @@ function Blend.Children(parent, value)
...
+	Cleanup:
+	* Attributes will be cleaned up on unsubscribe
+
+	@param parent Instance
+	@param value any
+	@return MaidTask
+]=]
+function Blend.Attributes(inst, value)
+	assert(typeof(inst) == "Instance", "Bad instance")
+	return Blend.ComputedPairs(value, function(attributeName, attributeValue, innerMaid)
```
Comment (Quenty) [inline src/blend/src/Shared/Blend/Blend.lua:601]: Might be good to error if attributeName is not a string.

File: src/blend/src/Shared/Blend/Blend.lua [hunk 3]
```diff
@@ -548,6 +575,48 @@ function Blend.Children(parent, value)
...
+--[=[
+	Mounts attributes to the parent and returns an object which will cleanup and delete
+	all attributes when removed.
+
+	```lua
+	local attribute = Blend.State("abc")
+	Blend.New "ScreenGui" {
+		Parent = game.Players.LocalPlayer.PlayerGui;
+		[Blend.Attributes] = {
+			Attribute1 = attribute;
```
Comment (Quenty) [inline src/blend/src/Shared/Blend/Blend.lua:587]: Unfortunately, ComputedPairs does not let us take in an [Observable] = [Observable] yet, which would make sense in certain situations.

I think this can be a future problem.

File: src/blend/src/Shared/Blend/Blend.lua [hunk 4]
```diff
@@ -548,6 +575,48 @@ function Blend.Children(parent, value)
...
+			Attribute2 = 123;
+		};
+	};
+	```
+	Cleanup:
+	* Attributes will be cleaned up on unsubscribe
```
Comment (Quenty) [inline src/blend/src/Shared/Blend/Blend.lua:593]: nit: would reword this to `Attributes will be set to nil on unsubscribe`.

File: src/blend/src/Shared/Blend/Blend.lua [hunk 5]
```diff
@@ -1,3 +1,4 @@
+---@diagnostic disable: redefined-local
```
Comment (Quenty) [inline src/blend/src/Shared/Blend/Blend.lua:1]: nit: We should not disable this redefined-local lint, and instead we should actually not redefine a local value.

File: src/blend/package.json [hunk 6]
```diff
@@ -21,7 +21,8 @@
   },
   "license": "MIT",
   "contributors": [
-    "Quenty"
+    "Quenty",
+    "Nightcycle"
```
Comment (Quenty) [inline src/blend/package.json:25]: Make sure you have an NPM account :p

**Comment by Quenty** [comment]
If you want to also update the `plugin/ui-converter-plugin` to support this, that would be neat, but not necessary.

**Review by Quenty — CHANGES_REQUESTED** [review]
Looks great overall, left a ton of nit-picks. If you don't want to address them, let me know, and I can do it. :p

File: src/blend/src/Shared/Blend/Blend.lua (hunk 1 above)
Comment (nightcycle) [reply src/blend/src/Shared/Blend/Blend.lua:607]: I'm still getting my sea legs with brio, so I left the if Brio.isBrio for now, but you're welcome to change it. I believe I've ported the rest of the code though!

**Comment by nightcycle** [comment]
> If you want to also update the `plugin/ui-converter-plugin` to support this, that would be neat, but not necessary.

I agree that's a good idea however after looking over the code I admit it unfortunately might be above my current ability level.

File: src/blend/src/Shared/Blend/Blend.lua (hunk 3 above)
Comment (nightcycle) [reply src/blend/src/Shared/Blend/Blend.lua:587]: In that case I've added a bit of extra documentation here to better convey that + an optional workaround for now.

File: src/blend/src/Shared/Blend/Blend.lua (hunk 1 above)
Comment (Quenty) [reply src/blend/src/Shared/Blend/Blend.lua:614]: Sounds good.

**Review by Quenty — APPROVED** [review]
